src/porepy/models/solution_strategy.py

```diff
@@ -131,26 +137,64 @@ def __init__(self, params: Optional[dict] = None):
         )
         """Time manager for the simulation."""
 
+        self.restart_options: dict = params.get(
+            "restart_options",
+            {
+                "restart": False,
+                # Boolean flag controlling whether restart is active. Internally
+                # assumed to be False.
+                "pvd_file": None,
+                # Path to pvd file collecting either multiple time steps (generated
+                # through pp.Exporter.write_pvd()); or a pvd file associated to a single
+                # time step (generated through pp.Exporter._export_mdg_pvd()).
+                "is_mdg_pvd": False,
+                # Boolean flag controlling whether prescribed pvd file is a mdg pvd
+                # file, i.e., created through Exporter._export_mdg_pvd(). Otherwise,
+                # it is assumed, the provided pvd file originates from
+                # Exporter.write_pvd(). If not provided, assumed to be False.
+                "vtu_files": None,
+                # Path(s) to vtu file(s), (alternative to pvd files which are preferred
+                # if present). Required if pvd_file is not specified.
+                "times_file": None,
+                # Path to json file containing evolution of exported time steps and
+                # used time step size at that time. If 'None' a default value,
+                # defined in :class:`~porepy.numerics.time_step_control.TimeManager
+                # is internally used.
+                "time_index": -1,
+                # Index addressing history in times_file; only relevant if "vtu_files"
+                # is not 'None' or "is_mdg_pvd" is 'True'. The index corresponds to
+                # the single time step vtu/pvd files. If not provided, internally
+                # assumed to address the last time step in times_file.
+            },
+        )
+        """Restart options (template) for restart from pvd as expected restart routines
+        within :class:`~porepy.viz.data_saving_model_mixin.DataSavingMixin`.
+        """
+
     def prepare_simulation(self) -> None:
         """Run at the start of simulation. Used for initialization etc."""
         # Set the geometry of the problem. This is a method that must be implemented
         # in a ModelGeometry class.
         self.set_geometry()
-        # Exporter initialization must be done after grid creation.
+
+        # Exporter initialization must be done after grid creation,
+        # but prior to data initialization.
+        self.initialize_data_saving()
 
         # Set variables, constitutive relations, discretizations and equations.
         # Order of operations is important here.
         self.set_equation_system_manager()
         self.set_materials()
         self.create_variables()
         self.initial_condition()
+        self.reset_state_from_file()
```

Comment:
@keileg I want to reopen a discussion I believe is closely related to some of your concerns you raised about the design. I want to stress the separation of initial_condition and restart. I believe in this way the user will not have to be careful in how to define initial conditions, though the user still needs to consider tailoring `reset_state_from_file` for his/her purposes.

Reply:
So, the idea is to always use the standard initialization, then always run the reset function, but leave it to the individual models (both standard and as changed by the user) to decide what to do? Then the standard models will presumably implement a reset method which does the natural things, and which can be activated by the reset parameters?

Reply:
I believe the answer here is yes, without anticipating everything possible.